tests/test_utils/tests.py

```diff
@@ -2186,6 +2186,8 @@ def test_allowed_database_copy_queries(self):
         finally:
             new_connection.validate_thread_sharing()
             new_connection._close()
+            if hasattr(new_connection, "close_pool"):
+                new_connection.close_pool()
```

Comment:
I think this should be in `_close()` for the Oracle backend instead :thinking:

Reply:
The `_close()` function only releases the connection back to the pool, it doesn't close the pool itself. This means that the connection remains valid and connected to the database unless we close the pool itself. As a result, dropping the user at the end of the test suite completion was causing the error "cannot drop a user who is already connected". This issue is also present in Postgres, as the `_close` function in Postgres also only releases the connection back to the pool. Therefore, this change fixes the issue for both backends.